### PR TITLE
[AC-2520] Remove Unassigned Items Banner

### PR DESCRIPTION
**Pull request by vincentsalucci**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/AC-2520

## 📔 Objective

> Remove all `Unassigned Items Banner` code

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

**Review by shane-melton — APPROVED** [review]
Vault change looks good!

**Review by willmartian — APPROVED** [review]

File: libs/common/src/platform/state/state-definitions.ts [hunk 1]
```diff
@@ -95,10 +95,6 @@ export const NEW_WEB_LAYOUT_BANNER_DISK = new StateDefinition("newWebLayoutBanne
   web: "disk-local",
 });
 
-export const UNASSIGNED_ITEMS_BANNER_DISK = new StateDefinition("unassignedItemsBanner", "disk", {
-  web: "disk-local",
-});
-
```
Comment (eliykat) [inline libs/common/src/platform/state/state-definitions.ts:101]: We need a data migration to remove this data at the same time as removing the StateDefinition.

See [this Slack conversation](https://bitwarden.slack.com/archives/C02LPDKPML6/p1712719872504199) for previous discussion on how to do this.

So you'll need to write a `Migrator` subclass in `libs/common/src/state-migrations/migrations` which gets that accounts object and removes this key from it.

**Review by eliykat — CHANGES_REQUESTED** [review]
Looks good, just gotta make sure we clean up our data on the way out 🧹

File: libs/common/src/platform/state/state-definitions.ts (hunk 1 above)
Comment (vincentsalucci) [reply libs/common/src/platform/state/state-definitions.ts]: Thanks for the reminder. I've reverted this change and will create a follow-up PR with the state-definition removal/migration.

**Comment by vincentsalucci** [comment]
@eliykat @willmartian @shane-melton : Reverted a change that is not relevant to your prior reviews 👍

**Review by shane-melton — APPROVED** [review]

**Review by willmartian — APPROVED** [review]

**Review by eliykat — APPROVED** [review]

**Pull request merged**